js/mushino.js

```diff
@@ -0,0 +1,672 @@
+            priceIsRequired = true;
+        }
+        if (priceIsRequired && price === undefined) {
+            throw new InvalidOrder ('createOrder method requires a price argument for a ' + type + ' order');
+        }
+        if (triggerPriceIsRequired && stopPrice === undefined) {
+            throw new InvalidOrder ('createOrder method requires a trigger_price as an extra param for a ' + type + ' order');
+        }
+        const response = await this[method] (this.extend (order, params));
+        return this.parseOrder (response['result']['order'], market);
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        const response = await this.privateGetBalances (this.extend (request, params));
+        const balances = response['result'];
+        const result = { 'info': balances };
+        const currencies = Object.keys (balances);
+        for (let i = 0; i < currencies.length; i++) {
+            const currencyId = currencies[i];
+            const balance = balances[currencyId];
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            const free_cash = this.safeFloat (balance, 'free');
+            const locked_cash = this.safeFloat (balance, 'locked');
+            const withdrawn_cash = this.safeFloat (balance, 'withdrawn');
+            const free_margin = this.safeFloat (balance, 'margin');
+            const free_total = free_cash + (free_margin || 0);
+            const used_total = locked_cash + withdrawn_cash;
+            const total = free_total + used_total;
+            account['free'] = free_total;
+            account['used'] = used_total;
+            account['total'] = total;
+            result[code] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostAuthCancel (this.extend ({
+            'id': id,
+        }, params));
+        return this.parseOrder (response['result']);
+    }
+
+    async cancelOrders (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostAuthCancelBulk (this.extend ({
+        }, params));
+        return this.parseOrders (response['result'], undefined, undefined, undefined);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        const id = this.safeString (transaction, 'id');
+        const address = this.safeString (transaction, 'address');
+        const tx = this.safeValue (transaction, 'transaction');
+        const txHash = this.safeString (tx, 'hash');
+        const timestamp = this.safeInteger (transaction, 'time');
+        const currencyId = this.safeString (transaction, 'asset');
+        const type = id.indexOf (currencyId) !== -1 ? 'deposit' : 'withdrawal';
```

Comment:
We need to split this line into several lines and add some brackets around the ternary conditional.